atecc608a_utils.c

```diff
@@ -0,0 +1,126 @@
+/**
+ * \file atecc608a_utils.c
+ * \brief ATECC508A and ATECC509A utility functions.
+ */
+
+/*
+ *  Copyright (C) 2019, ARM Limited, All Rights Reserved
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+#include "atecc608a_se.h"
+#include "atecc608a_utils.h"
+
+#include "atca_basic.h"
+
+ATCAIfaceCfg atca_iface_config = {
+    .iface_type = ATCA_I2C_IFACE,
+    .devtype = ATECC608A,
+    .atcai2c.slave_address = 0xC0,
+    .atcai2c.bus = 2,
+    .atcai2c.baud = 400000,
+    .wake_delay = 1500,
+    .rx_retries = 20,
+};
+
+psa_status_t atecc608a_to_psa_error(ATCA_STATUS ret)
```

Comment:
We don't need two copies of this function in different code bases. https://github.com/ARMmbed/mbed-os-atecc608a/pull/3 should expose this function, since it's convenient to have it outside the driver.

On a platform with secure/non-secure isolation, the application and the driver will be in separate linker namespaces, so the function won't be accessible in the application. But the application wouldn't work anyway, since it wouldn't be able to access the hardware.

Same remark for other support functions and macros like `ATCAB_INIT` and friends.

Reply:
Added.